src/client.ts

```diff
@@ -8805,8 +8805,8 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
         ruleId: Exclude<string, RuleId>,
         body: Pick<IPushRule, "actions" | "conditions" | "pattern">,
     ): Promise<{}> {
-        // NB. Scope not uri encoded because devices need the '/'
-        const path = utils.encodeUri("/pushrules/" + scope + "/$kind/$ruleId", {
+        const path = utils.encodeUri("/pushrules/$scope/$kind/$ruleId", {
+            $scope: scope,
```

Comment:
I don't really see that this is a worthwhile improvement. It doesn't actually *remove* the redundant `scope` param (which would be a  concrete improvement, though hard to do without breaking backwards compatibility); rather it just url-encodes it. Since `scope` must always be `global`, this seems to make no difference at all?

Reply:
I agree. It is a miniscule improvement to legibility / removing redundant comments which may confuse, and the PR description echoes your sentiment about preserving backwards compatibility.

Was hoping that a small touch-up would be appreciated, but if you prefer to keep things as they are, I'm content to cancel this PR as well.

Reply:
Yeah, let's leave it; I don't think it's worthwhile.

Maybe in future we can create `addGlobalPushRule` etc which omit the `scope` param; then we can deprecate the existing methods.